api/auth/gcp/gcp.go

```diff
@@ -0,0 +1,184 @@
+package gcp
```

Comment:
It would be cool to eventually replace the login helper (see [cli.go](https://github.com/hashicorp/vault-plugin-auth-gcp/blob/master/plugin/cli.go)) in the plugin with this code. It'd make it quick to add support for obtaining GCE tokens from within GCE instances 👍